webapp/src/Service/ImportExportService.php

```diff
@@ -251,6 +252,22 @@ public function importContestData(mixed $data, ?string &$errorMessage = null, st
         }
 
         $this->em->persist($contest);
+
+        $shadow = $data['shadow'] ?? null;
+        if ($shadow) {
+            $externalSource = $this->em->getRepository(ExternalContestSource::class)->findOneBy(['contest' => $contest]) ?: new ExternalContestSource();
+            $externalSource->setContest($contest);
+            foreach (['source', 'username', 'password', 'type'] as $field) {
+                // Overwrite the existing value if the property is defined in the data: $externalSource-setSource($data['shadow']['source'])
+                $fieldFunc = 'set'.ucwords($field);
+                $fieldArgs = [$shadow[$field]];
+                if (isset($shadow[$field])) {
+                    $externalSource->$fieldFunc(...$fieldArgs);
+                }
+            }
```

Comment:
```suggestion
            foreach ($shadow as $field => $value) {
                $fieldFunc = 'set'.ucwords($field);
                $fieldArgs = [$value];
                if (method_exists($externalSource, $fieldFunc)) {
                    $externalSource->$fieldFunc(...$fieldArgs);
                }
            }
```

would now also work right? That way if we ever add fields it just works ™️